callbacks/agent_final_stream.go

```diff
@@ -54,16 +59,41 @@ func (handler *AgentFinalStreamHandler) GetEgress() chan []byte {
 // The callback function receives two parameters:
 // - ctx: the context.Context object for the egress operation.
 // - chunk: a byte slice representing a chunk of data from the egress channel.
+// returns a pointer to a sync.WaitGroup object that is used to wait from the caller.
 func (handler *AgentFinalStreamHandler) ReadFromEgress(
 	ctx context.Context,
 	callback func(ctx context.Context, chunk []byte),
-) {
+) *sync.WaitGroup {
+	wg := &sync.WaitGroup{}
+	wg.Add(1)
 	go func() {
-		defer close(handler.egress)
-		for data := range handler.egress {
-			callback(ctx, data)
+		defer wg.Done()
+
+	FORLOOP:
```

Comment:
it's fairly uncommon to break to a label like this, can you consider alternatives here?

Reply:
Yes, use return is more concisely than break label